experimental/promptbuilder/prompt_builder.py

```diff
@@ -0,0 +1,297 @@
+from typing import List, Optional, Type, Union
+from enum import Enum, auto
+from pydantic.dataclasses import dataclass
+from pydantic import BaseModel, Field, create_model
+import openai
+
+from openai_function_call import OpenAISchema
+
+
+class MessageRole(Enum):
+    USER = auto()
+    SYSTEM = auto()
+    ASSISTANT = auto()
+
+
+@dataclass
```

Comment:
is there a reason for using dataclasses here instead of subclassing BaseModel? (Not asking a a review, just for learning)

I would assume b/c of post_init, but is that really necessary if the `Message` class itself is only used as base class?

Reply:
@jxnl Hi Jason, first, thank you for sharing your experience and wisdom, second, thank you again. 

Comment only from a novice, newbie, beginner's perspective - For examples for "functions" provide dynamic examples that iterates/finds the right function out of possibly hundreds of functions to send to ChatGPT, otherwise stuffing all those functions will reach content limit. I tried samantic search using pinecone but was a miss and a hit